src/java.base/share/classes/javax/crypto/spec/HKDFParameterSpec.java

```diff
@@ -0,0 +1,457 @@
+         * a hardware crypto device. This method appends to the existing list of
+         * values or creates a new list if there are none yet.
+         * <p>
+         * This supports the use-case where a label can be applied to the IKM
+         * but the actual value of the IKM is not yet available.
+         *
+         * @param ikm
+         *     the initial key material value
+         *
+         * @return this builder
+         *
+         * @throws NullPointerException
+         *     if the {@code ikm} is null or empty
+         */
+        public Builder addIKM(byte[] ikm) {
+            if (ikm != null && ikm.length != 0) {
+                return addIKM(new SecretKeySpec(ikm, "Generic"));
+            } else {
+                throw new NullPointerException("ikm must not be null or empty");
+            }
+        }
+
+        /**
+         * {@code addSalt} may be called when the salt value is to be assembled
+         * piece-meal or if part of the salt is to be supplied by a hardware
+         * crypto device. This method appends to the existing list of values or
+         * creates a new list if there are none yet.
+         * <p>
+         * This supports the use-case where a label can be applied to the salt
+         * but the actual value of the salt is not yet available.
+         *
+         * @param salt
+         *     the salt value
+         *
+         * @return this builder
+         *
+         * @throws NullPointerException
+         *     if the {@code salt} is null
+         */
+        public Builder addSalt(SecretKey salt) {
+            if (salt != null) {
+                salts.add(salt);
+            } else {
+                throw new NullPointerException("salt must not be null");
+            }
+            return this;
+        }
+
+        /**
+         * {@code addSalt} may be called when the salt value is to be assembled
+         * piece-meal or if part of the salt is to be supplied by a hardware
+         * crypto device. This method appends to the existing list of values or
+         * creates a new list if there are none yet.
+         * <p>
+         * This supports the use-case where a label can be applied to the salt
+         * but the actual value of the salt is not yet available.
+         *
+         * @param salt
+         *     the salt value
+         *
+         * @return this builder
+         *
+         * @throws NullPointerException
+         *     if the {@code salt} is null or empty
+         */
+        public Builder addSalt(byte[] salt) {
+            if (salt != null && salt.length != 0) {
+                return addSalt(new SecretKeySpec(salt, "Generic"));
+            } else {
+                throw new NullPointerException(
+                    "salt must not be null or empty");
+            }
+        }
+    }
+
+    /**
+     * Static helper-method that may be used to initialize a {@code Builder}
+     * with an empty {@code Extract}
```

Comment:
No need to say it is static. I would also avoid terms like "helper-method". Suggest rewording as "Returns a builder for building `Extract` and `ExtractExpand` objects.